cfme/rest/__init__.py

```diff
@@ -339,3 +341,28 @@ def _finished():
             collection.action.delete(*delete_entities)
 
     return entities
+
+
+def templates(request, rest_api, a_provider, vm, num=1):
+    vm_for_template = VM.factory(vm, a_provider)
+    template_names = []
+    for _index in range(0, num):
+        name = "test_rest_template_{}_{}".format(_index, fauxfactory.gen_alphanumeric())
+        template_names.append(name)
+        vm_for_template.publish_to_template(name)
+
+    tpls = []
+    for tpl in template_names:
+        wait_for(lambda: rest_api.collections.templates.find_by(name=tpl), num_sec=180, delay=10)
+        tpls.append(rest_api.collections.templates.get(name.tpl))
+
+    @request.addfinalizer
+    def _finished():
+        ids = [t.id for t in tpls]
+        delete_tpls = [t for t in rest_api.collections.templates if t.id in ids]
+        if len(delete_tpls) != 0:
+            rest_api.collections.templates.action.delete(*delete_tpls)
+
+    if num == 1:
```

Comment:
@RonnyPfannschmidt 
Agree, however I think it will be better to return tpls[-1]?